File: trunk/data/modules/trading.c
module trading {
  import random;
  int quantity;
  int last_ship;
  float price_instability;
  void init(){
    lasttime=0.0;
    last_ship=0;
    price_instability=0.01;
    quantity=4;
  };
  void SetPriceInstability(float inst) {
    price_instability=inst
  }
  void SetMaxQuantity (int quant) {
    quantity=quant;
  }
  void trade_cargo(){
    int incdec_ships=random.randomint(1,2);
    int quant = random.randomint (1,quantity);
    object un = _unit.getUnit (last_ship);
    if (_std.isNull(un)) {
      last_ship=0;
    } else {
      object player = _unit.getPlayer();
      if (player!=un) {
	if (incdec_ships==1) {
	  _unit.incrementCargo(un,1-(quant*price_instability),quant);
	}else {
	  _unit.decrementCargo(un,1+(quant*price_instability),quant);
	}
      }
      last_ship=last_ship+1;
    }
  };
}

File: trunk/data/modules/trading.py
import vsrandom
import VS
class trading:
  def __init__(self):
    self.last_ship=0
    self.quantity=4
    self.price_instability=0.01
    
  def SetPriceInstability(self, inst):
    self.price_instability=inst
  
  def SetMaxQuantity (self,quant):
    self.quantity=quant
  
  def Execute(self):
    quant = (vsrandom.random()*(self.quantity-1))+1
    un = VS.getUnit (self.last_ship)
    if (un.isNull()):
      self.last_ship=0
    else:
      if (un.isSignificant()):
	player = VS.getPlayer()
	if (player!=un):
	  if (vsrandom.random()<.5):
	    un.incrementCargo(1-(quant*self.price_instability),quant)
	  else:
	    un.decrementCargo(1+(self.price_instability))
      self.last_ship+=1
  
